app/src/main/java/org/sagebionetworks/research/mpower/medication/MedicationAdapter.kt

```diff
@@ -0,0 +1,193 @@
+package org.sagebionetworks.research.mpower.medication
+
+import android.app.TimePickerDialog
+import android.content.Context
+import android.support.v7.widget.RecyclerView
+import android.text.Editable
+import android.text.TextWatcher
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import kotlinx.android.synthetic.main.medication_add.view.*
+import kotlinx.android.synthetic.main.medication_dosage.view.*
+import kotlinx.android.synthetic.main.medication_schedule.view.*
+import org.sagebionetworks.research.mpower.R.layout
+import org.sagebionetworks.research.mpower.medication.Type.ADD
+import org.sagebionetworks.research.mpower.medication.Type.DOSAGE
+import org.sagebionetworks.research.mpower.medication.Type.SCHEDULE
+import org.slf4j.LoggerFactory
+import java.text.SimpleDateFormat
+import java.util.Calendar
+
+class MedicationAdapter(var name: String, var items : List<MedicationItem>, val context: Context, val listener: Listener)
+    : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
+
+    private val LOGGER = LoggerFactory.getLogger(
+            MedicationAdapter::class.java)
+
+    override fun getItemCount(): Int {
+        return items.size
+    }
+
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
+        lateinit var holder: RecyclerView.ViewHolder
+        when(viewType) {
+            0 ->
```

Comment:
Should this when statement use DOSAGE, ADD and SCHEDULE?

Reply:
Not directly but I should create a mapping between the enum and the view type and eliminate these hardcoded ints.